### PR TITLE
Adds YouTube iFrame for each lecture

### PR DESCRIPTION
**Pull request by jaysaleh**
This pull request adds a YouTube iFrame to `lecture-view.html`

### CODE REVIEW

File: src/main/webapp/lectures.js [hunk 1]
```diff
@@ -1,5 +1,6 @@
 /* Used to gather URL parameters. */
 const URL_PARAM_ID = 'id';
+const URL_PARAM_VIDEO_ID = 'videoId';
```
Comment (n-wach) [inline src/main/webapp/lectures.js:3]: `'video-id'` to match other url/request params?

File: src/main/webapp/view.js [hunk 2]
```diff
@@ -1,11 +1,57 @@
...
 
 /** Returns database ID for lecture. */
 function getLectureKey() {
   const urlParams = new URLSearchParams(window.location.search);
-  return urlParams.get('id');
+  return urlParams.get(URL_PARAM_ID);
+}
+
+/** Returns video ID for lecture. */
+function getVideoId() {
```
Comment (n-wach) [inline src/main/webapp/view.js:54]: I thought you were going to query the LectureServlet with the lecture id to get the video id

Comment (n-wach) [reply src/main/webapp/view.js:54]: Rather than also have the video id in the url

File: src/main/webapp/view.js [hunk 3]
```diff
@@ -1,11 +1,57 @@
+/* Used to gather URL parameters. */
```
Comment (n-wach) [inline src/main/webapp/view.js:1]: I think most of this stuff belongs in it's own `video.js` file

File: src/main/webapp/view.js (hunk 2 above)
Comment (jaysaleh) [reply src/main/webapp/view.js:54]: Isn't it more efficient this way?

File: src/main/webapp/view.js (hunk 3 above)
Comment (jaysaleh) [reply src/main/webapp/view.js]: fixed

File: src/main/webapp/lectures.js (hunk 1 above)
Comment (jaysaleh) [reply src/main/webapp/lectures.js]: fixed

File: src/main/webapp/lecture-view.html [hunk 4]
```diff
@@ -3,5 +3,5 @@
 
 <title>ZoomTube</title>
 <link rel="stylesheet" href="style.css" />
-<script src="view.js" defer></script>
-<p>View Lecture Page</p>
+<script src="video.js" defer></script>
```
Comment (n-wach) [inline src/main/webapp/lecture-view.html:6]: We should load both the view and video scripts.  View is the one that parses the lecture id and initializes the other sections
```suggestion
<script src="view.js" defer></script>
<script src="video.js" defer></script>
```

File: src/main/webapp/video.js [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+  });
+}
+
+/** Plays video. Called after API and iFrame load. */
+function onPlayerReady(event) {
+  event.target.playVideo();
+}
+
+/** Returns database ID for lecture. */
+function getLectureKey() {
```
Comment (n-wach) [inline src/main/webapp/video.js:50]: This should stay in view.js, maybe also create a loadVideo() in here that view.js calls in `initialize`.  Sorry for being unclear, take a look at master to see how discussion.js and view.js interact, maybe merge them?

Comment (n-wach) [reply src/main/webapp/video.js:50]: As in merge master into this branch

File: src/main/webapp/lecture-view.html (hunk 4 above)
Comment (jaysaleh) [reply src/main/webapp/lecture-view.html:6]: fixed

**Comment by jaysaleh** [comment]
Reopening in a different branch

**Pull request closed without merging**